doc/conf.py

```diff
@@ -67,26 +97,52 @@ def run_doxygen():
         os.chdir(curdir)
 
 
-def build_jvm_docs():
-    """Build docs for the JVM packages"""
+def get_branch() -> str:
+    """Guess the git branch."""
     git_branch = os.getenv("READTHEDOCS_VERSION_NAME", default=None)
     print(f"READTHEDOCS_VERSION_NAME = {git_branch}")
 
-    if not git_branch:
-        git_branch = "master"
+    if not git_branch:  # Not in RTD
+        git_branch = "master"  # use the master branch as the default.
     elif git_branch == "latest":
         git_branch = "master"
     elif git_branch == "stable":
         git_branch = f"release_{xgboost.__version__}"
+    else:  # Likely PR branch
+        git_branch = f"PR-{git_branch}"
     print(f"git_branch = {git_branch}")
+    return git_branch
+
+
+def get_sha(branch: str) -> str | None:
+    sha = os.getenv("READTHEDOCS_GIT_COMMIT_HASH", default=None)
+    if sha is not None:
+        return sha
+
+    if branch == "master":
+        res = subprocess.run(["git", "rev-parse", "master"], stdout=subprocess.PIPE)
+    else:
```

Comment:
Do we need to check release branch?

Reply:
Not sure why it's there

Reply:
We need to get the commit hash for the correct artifact. Otherwise we need to specialize the GitHub workflow for release branch